File: Dante2/Module2.04/Zadanie4.06/main.c
#include <stdio.h>
#include <stdlib.h>
#include "array.h"

int main()
{
    int num;

    printf("Enter: ");
    int check = scanf("%i",&num);
    if(!check)
    {
        printf("Incorrect input");
        return 1;
    }
    if(num == 0)
    {
        printf("Incorrect input data");
        return 2;
    }

    struct array_t a;
    check = array_create(&a, num);
    if(check == 1)
    {
        printf("Incorrect input data");
        return 2;
    }
    if(check == 2)
    {
        printf("Failed to allocate memory");
        return 8;
    }

    int number = 1;
    int pointer = 0;
    printf("Enter: ");

    while(number != 0)
    {
        check = scanf("%i",&number);
        if(!check)
        {
            printf("Incorrect input");
            array_destroy(&a);
            return 1;
        }
        if(number != 0)
        {
            pointer = 1;
        }
        check = array_push_back(&a, number);
        if(check == 1)
        {
            printf("Incorrect input data");
            return 2;
        }
        if(check == 2)
        {
            break;
        }
    }
    if(pointer == 0)
    {
        printf("Buffer is empty\n");
        array_destroy(&a);
        return 0;
    }
    if(a.size == a.capacity)
    {
        printf("Buffer is full\n");
    }

    array_display(&a);
    array_destroy(&a);

    return 0;
}

File: Dante2/Module2.04/Zadanie4.06/main.rs
mod custom_array;
fn main() {
    let mut input: String = String::new();
    println!("Enter: ");

    std::io::stdin().read_line(&mut input).expect("Incorrect input");
    let size: usize = input.trim().parse().expect("Incorrect parse");

    if size == 0 {
        panic!("Incorrect input data");
    }

    let mut a: custom_array::ArrayT = Default::default();
    custom_array::array_create(&mut a, size);

    println!("Enter: ");
    loop {
        input.clear();
        std::io::stdin().read_line(&mut input).expect("Incorrect input");
        let number: i32 = input.trim().parse().expect("Incorrect parse");

        if number == 0 {
            break;
        }

        match custom_array::array_push_back(&mut a, number) {
            1 => panic!("Incorrect input data"),
            2 => break,
            _other => continue
        }
    }

    if a.size == 0 {
        println!("Buffer is empty");
        return;
    }
    if a.size == a.capacity {
        println!("Buffer is full");
        return;
    }

    custom_array::array_display(&a);
}
